model/client4.go

```diff
@@ -36,10 +37,15 @@ func NewAPIv4Client(url string) *Client4 {
 }
 
 func BuildErrorResponse(r *http.Response, err *AppError) *Response {
-	if r == nil {
-		return &Response{StatusCode: 0, Error: err}
-	} else {
-		return &Response{StatusCode: r.StatusCode, Error: err}
+	statusCode := 0
+	if r != nil {
+		statusCode = r.StatusCode
+	}
+
+	return &Response{
+		StatusCode: statusCode,
+		Error:      err,
+		Response:   r,
```

Comment:
Yeah, maintaining compatibility would be important. And changing it to only return the header field eliminates that redundancy as well. :)